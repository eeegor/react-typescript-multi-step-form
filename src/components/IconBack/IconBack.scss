.icon-back {
	polygon {
		transition: all 0.3s ease-in-out;
	}
}
