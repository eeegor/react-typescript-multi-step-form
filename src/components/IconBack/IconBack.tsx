/**
 * Component IconBack
 */

import React from 'react';
import classnames from 'classnames';
import './IconBack.scss';

export interface Props {
	id: string;
	className?: string;
	color?: string;
}

export const IconBack = (props: Props): JSX.Element => {
	const { id, className, color } = props;

	return (
		<svg
			id={id}
			className={classnames('icon-back', className)}
			version="1.1"
			viewBox="0 0 512 512"
			height="36"
			width="36"
			xmlSpace="preserve"
			xmlns="http://www.w3.org/2000/svg"
			xmlnsXlink="http://www.w3.org/1999/xlink"
		>
			<polygon
				fill={color || '#333'}
				points="352,128.4 319.7,96 160,256 160,256 160,256 319.7,416 352,383.6 224.7,256 "
			/>
		</svg>
	);
};
