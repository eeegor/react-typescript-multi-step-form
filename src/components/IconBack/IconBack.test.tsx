import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { IconBack } from './IconBack';
import { mount } from 'enzyme';

describe('IconBack', () => {
	const callback = jest.fn();

	it('renders markup correctly', () => {
		const wrapper = <IconBack id="next" />;
		const tree = renderer.create(wrapper).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
