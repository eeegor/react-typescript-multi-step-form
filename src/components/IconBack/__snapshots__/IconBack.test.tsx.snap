// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconBack renders markup correctly 1`] = `
<svg
  className="icon-back"
  height="36"
  id="next"
  version="1.1"
  viewBox="0 0 512 512"
  width="36"
  xmlSpace="preserve"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <polygon
    fill="#333"
    points="352,128.4 319.7,96 160,256 160,256 160,256 319.7,416 352,383.6 224.7,256 "
  />
</svg>
`;
